sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala

```diff
@@ -662,7 +662,7 @@ case class AssertNotNull(child: Expression, walkedTypePath: Seq[String])
   extends UnaryExpression with NonSQLExpression {
 
   override def dataType: DataType = child.dataType
-
+  override def foldable: Boolean = false
```

Comment:
Currently, `AssertNotNull.eval` throws `UnsupportedOperationException` if it has a foldable child.
I hope to improve that in another PR.